algorithms/pigeonholeSort.py

```diff
@@ -1,20 +1,34 @@
+# for pigeonholes, the indexes represent:
+# the value from the array - the minimum value,
+# and the item is the number of times that value is in the array
+# we initialize the counts to 0
+def pigeonholes(size):
+  return [0]*size
+
+def addTo(array, value):
+  array[value] += 1
+
+def subtractFrom(array, value):
+  array[value] -= 1
+
 def pigeonholeSort(array, *args):
-  minV = min(array)
-  maxV = max(array)
-  A = array.copy()
-  size = maxV - minV + 1
-  C = [0]*(size)
-  i = 0
-  for x in array:
-    yield A, i, -1, -1, -1
-    C[x - minV] += 1
-    A[i] = A[i] / (C[x-minV] + 1)
-    i += 1
-  i = 0
-  for count in range(size):
-    while C[count] > 0:
-      yield A, i-1, -1, C[count] * minV , -1
```

Comment:
Note: `yield` calls the `drawBars` method and passes these arguments to it.  `-1` parameters hide the corresponding `redBar`s and `blueBar`s.  The `blueBar1` argument didn't seem to do anything that made sense so I removed it.

see: https://github.com/JasonOna/Sorting-Algorithms-Visualizer/blob/0059fc66362fd8f3c8b3e5404e1977ca3e7932db/display.py#L234